### PR TITLE
Update Kit to use latest Apptentive SDK 5.1.4

### PR DESCRIPTION
**Pull request by weeeBox**
Bug fixes + mParticle person id support

### CODE REVIEW

File: src/main/java/com/mparticle/kits/ApptentiveKit.java [hunk 1]
```diff
@@ -40,11 +55,30 @@ public String getName() {
...
 			throw new IllegalArgumentException("Apptentive App Signature is required. If you are migrating from a previous version, you may need to enter the new Apptentive App Key and Signature on the mParticle website.");
 		}
-		ApptentiveConfiguration configuration = new ApptentiveConfiguration(apptentiveAppKey, apptentiveAppSignature);
+
+    ApptentiveConfiguration configuration = new ApptentiveConfiguration(apptentiveAppKey, apptentiveAppSignature);
 		Apptentive.register((Application)context.getApplicationContext(), configuration);
+
+		Activity activity = getActivity();
+		if (activity != null) {
+			ApptentiveActivityLifecycleCallbacks.getInstance().onActivityCreated(activity, null);
```
Comment (samdozor) [inline src/main/java/com/mparticle/kits/ApptentiveKit.java]: this shouldn't be necessary - our KitManager replays these lifecycle events automatically.

File: src/main/java/com/mparticle/kits/ApptentiveKit.java [hunk 2]
```diff
@@ -40,11 +55,30 @@ public String getName() {
...
+		ApptentiveNotificationCenter.defaultCenter()
+				.addObserver(ApptentiveNotifications.NOTIFICATION_CONVERSATION_STATE_DID_CHANGE, this);
+
 		return null;
 	}
 
+	@Override
+	protected void onKitDestroy() {
+		super.onKitDestroy();
+
```
Comment (samdozor) [inline src/main/java/com/mparticle/kits/ApptentiveKit.java]: please remove extra line break.

File: src/main/java/com/mparticle/kits/ApptentiveKit.java (hunk 1 above)
Comment (weeeBox) [reply src/main/java/com/mparticle/kits/ApptentiveKit.java]: @samdozor, our SDK requires to be registered in the `Application.onCreate` method where it subscribes for the activity lifecycle callbacks. If we miss any of the the activity lifecycle's events - we're not gonna be able to detect background/foreground events correctly.

Comment (weeeBox) [reply src/main/java/com/mparticle/kits/ApptentiveKit.java]: @samdozor, let me run a few tests really quick to see if our SDK still works as intended.

Comment (weeeBox) [reply src/main/java/com/mparticle/kits/ApptentiveKit.java]: @samdozor, I've verified the behavior. Everything works fine after removing manual activity lifecycle calls. Thanks for pointing that out!

**Comment by weeeBox** [comment]
@samdozor, is there anything else which we should address before merging the PR?

**Pull request merged**